arangod/RocksDBEngine/RocksDBBuilderIndex.cpp

```diff
@@ -167,24 +437,79 @@ Result RocksDBBuilderIndex::remove(transaction::Methods& trx,
   return Result();  // do nothing
 }
 
+static Result processPartitions(
+    bool isForeground, std::deque<std::pair<uint64_t, uint64_t>> partitions,
+    trx::BuilderTrx& trx, rocksdb::Snapshot const* snap,
+    RocksDBCollection* rcoll, rocksdb::DB* rootDB, RocksDBIndex& ridx,
+    std::atomic<uint64_t>& docsProcessed, size_t numThreads,
+    uint64_t threadBatchSize, rocksdb::Options const& dbOptions,
+    std::string const& idxPath) {
+  auto sharedWorkEnv =
+      std::make_shared<SharedWorkEnv>(std::move(partitions), rcoll->objectId());
+  std::vector<std::unique_ptr<IndexCreatorThread>> idxCreatorThreads;
+  for (size_t i = 0; i < numThreads; ++i) {
+    auto newThread = std::make_unique<IndexCreatorThread>(
+        false, isForeground, threadBatchSize, docsProcessed, sharedWorkEnv,
+        rcoll, rootDB, ridx, snap, dbOptions, idxPath);
+    idxCreatorThreads.emplace_back(std::move(newThread));
+  }
+
+  try {
+    for (auto& idxCreatorThread : idxCreatorThreads) {
+      idxCreatorThread->start();
+    }
+  } catch (std::exception const& ex) {
+    LOG_TOPIC("01ad6", WARN, Logger::ENGINES)
+        << "error while starting index creation thread: " << ex.what();
+    // abort the startup
+    sharedWorkEnv->registerError({TRI_ERROR_INTERNAL, ex.what()});
+  }
+  sharedWorkEnv->waitUntilAllThreadsTerminate();
+
+  uint64_t seekCounter = 2;
+  uint64_t nextCounter = 0;
+  for (auto const& threadStats : sharedWorkEnv->getThreadStatistics()) {
+    seekCounter += threadStats.numSeeks;
+    nextCounter += threadStats.numNexts;
+  }
+  LOG_TOPIC("d9bf2", DEBUG, Logger::ENGINES)
+      << "Total seeks: " << seekCounter << ", next: " << nextCounter;
```

Comment:
Can we add some context to this log message. Otherwise it is unclear that the seeks and next calls are originating from an index creation. Any context here would improve the log message.

Reply:
done